pkg/proxy/service.go

```diff
@@ -109,10 +109,12 @@ func (bsvcPortInfo *BaseServicePortInfo) ExternalIPStrings() []string {
 }
 
 // LoadBalancerIPStrings is part of ServicePort interface.
-func (bsvcPortInfo *BaseServicePortInfo) LoadBalancerIPStrings() []string {
+func (bsvcPortInfo *BaseServicePortInfo) LoadBalancerVIPStrings() []string {
 	var ips []string
 	for _, ing := range bsvcPortInfo.loadBalancerStatus.Ingress {
-		ips = append(ips, ing.IP)
+		if proxyutil.IsVIPMode(ing) {
+			ips = append(ips, ing.IP)
```

Comment:
So I noticed recently that `pkg/proxy/service.go` is sort of dumb about LoadBalancerIPs: it copies the entire `LoadBalancerStatus` object into `BaseServicePortInfo`, but then only allows callers to access the individual IPs. So if you're changing this code anyway, can you change `BaseServicePortInfo.loadBalancerStatus v1.LoadBalancerStatus` to `BaseServicePortInfo.loadBalancerVIPs []string` and do the filtering by `IsVIPMode` in `newBaseServiceInfo` rather than here?